specs/simple-serialize.md

```diff
@@ -54,43 +54,91 @@ For convenience we alias:
 
 We recursively define the `serialize` function which consumes an object `value` (of the type specified) and returns a bytestring of type `"bytes"`.
 
-*Note*: In the function definitions below (`serialize`, `hash_tree_root`, `signed_root`, etc.) objects implicitly carry their type.
+> *Note*: In the function definitions below (`serialize`, `hash_tree_root`, `signed_root`, `is_fixed_size`, `is_variable_size` etc.) objects implicitly carry their type.
 
-### `"uintN"`
+### Basic Types
+
+For basic types the `serialize` function is defined as follows.
+
+#### `"uintN"`
+
+A byte string of width  `N // 8` containing the little-endian encode integer.
 
 ```python
 assert N in [8, 16, 32, 64, 128, 256]
 return value.to_bytes(N // 8, "little")
 ```
 
-### `"bool"`
+#### `"bool"`
+
+* The byte `\x00` **if** the value is `False`
+* The byte `\x01` **if** the value is `True`
 
 ```python
 assert value in (True, False)
 return b"\x01" if value is True else b"\x00"
 ```
 
-### Vectors, containers, lists
+### Composite Types (Vectors, Containers and Lists)
 
-If `value` is fixed-size:
+The serialized representation of composite types is comprised of two binary sections.
 
-```python
-return "".join([serialize(element) for element in value])
-```
+* The first section is *fixed size* for all types, containing the concatenation of
```

Comment:
```suggestion
* The first section contains the concatenation of
```

Unless I'm misunderstanding something, lists would have a variable size first section.

Reply:
Yes, this was a carry over from before I merged the offsets into the first section, will update